bpf/bpf_host.c

```diff
@@ -188,6 +188,34 @@ ipv6_host_policy_egress(struct __ctx_buff *ctx, __u32 srcID)
 	if (hdrlen < 0)
 		return hdrlen;
 	l4_off = l3_off + hdrlen;
+
+#ifdef ENABLE_SERVICES
```

Comment:
Can we drop the service handling? Asking, as this implementation is not enough to properly handle a reverse path (e.g. reply coming from a pod running in the host netns (e.g. kube-apiserver) won't hit bpf_host.o, also a reply from a local service endpoint is not rev-DNAT xlated).

It's safe enough to assume that a user will either be running with bpf_sock or have kube-proxy (in both cases the host fw should see a real endpoint IP addr instead of VIP).

Reply:
> Can we drop the service handling? Asking, as this implementation is not enough to properly handle a reverse path (e.g. reply coming from a pod running in the host netns (e.g. kube-apiserver) won't hit bpf_host.o, also a reply from a local service endpoint is not rev-DNAT xlated).

Sure, less complexity issues. I'll remove commits `bpf: Support LB on host egress` and `daemon: Disable egress LB if EnableHostFirewall on older kernels` on the next iteration in the evening. I'll also disable service handling in BPF with a warning if the user enabled the host firewall.

Reply:
I removed those commits and, in commit `daemon: Introduce enable-host-firewall option`, I disabled BPF-based LB handling with a warning when the host firewall is enabled.